packages/translate-results/lib/__tests__/progress.test.ts

```diff
@@ -0,0 +1,40 @@
+import {translateProgress} from '../progress';
+
+describe('Interactive progress explanation.', () => {
+	describe('Test actionFailed status with codes.', () => {
+		it('Should return empty result if code not provided', () => {
+			expect(translateProgress({status: 'actionFailed'}))
+				.toStrictEqual({title: ''});
+		});
+
+		it('Should return empty result if code is unknown', () => {
+			expect(translateProgress({status: 'actionFailed', code: 'unknownCode' as any}))
+				.toStrictEqual({title: ''});
+		});
+
+		it('Should return translation if code is valid', () => {
+			expect(translateProgress({code: 'noActivePlan'}))
+				.toStrictEqual({
+					title: 'Cannot continue: Your subscription has expired',
+					description: 'Your subscription has expired, to continue using Suitest please [renew your subscription](https://the.suite.st/preferences/billing)',
+				});
+		});
+	});
+
+	describe('Test results with', () => {
+		it('status "nothing" should be empty string.', () => {
+			expect(translateProgress({status: 'nothing'}))
+				.toStrictEqual({title: ''});
+		});
+
+		it('unknown status should be empty string.', () => {
+			expect(translateProgress({status: 'unknownStatus' as any}))
+				.toStrictEqual({title: ''});
```

Comment:
Not sure about this one. Should we perhaps use generic text in this case?

Reply:
I think that is ok if we will display nothing if to us will come progress status which we cant translate.

Reply:
I would personally prefer if method returns or throws some error in this case.

Reply:
what about returning something like [error, translation] ?

Reply:
I think I would throw error or return undefined instead of object.
Then report it to sentry.